### PR TITLE
DYN-6527: Fix graph update for primitive input nodes that are first initialized to null 

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

Dropdown nodes (inheriting from `DropDownBase`) that produce primitive inputs were affected by changes done in #12835 where if `SelectedIndex` was set to -1 (the default value), then when the node was updated to a different dropdown enum value, it wouldn't update. The node would continue to return `null` in this case as the `SelectedIndex` value of -1 builds a null node.

This issue is not specific to dropdown node and can happen when primitive input ASTs (e.g. strings) are constructed in the following way:
Case 1:
```
var_a = null;
var_b = var_a;
...
// update var_b
var_b = "new string"; // var_b does not update
```
as opposed to this case where it works:
Case 2:
```
var_b = "old string";
....
// update
var_b = "new string"; // var_b updates
```

If I change the default value of `SelectedIndex` in `DropdownBase` to 0, such that a `null `AST is not first constructed, the node update works as expected.

~~WIP: Trying to work on a fix to handle the case where the input variable (graphnode) is first assigned to another variable that is `null`, and is later updated to a new primitive value.~~

In case 1, the instructions generated look like:
```
==============Start Node==============
[a.0.1414]push null
[a.0.1415]pop %t_39_01312cae28ba4c749d09ffe9d1be6e4f
[a.0.1416]dep 428[ExprUID] True[SSA]
==============End Node==============
==============Start Node==============
[a.0.1417]push %t_39_01312cae28ba4c749d09ffe9d1be6e4f
[a.0.1418]pop _lx
[a.0.1419]push _lx
[a.0.1420]pop var_22759f72156c4b68930de54494586bd4
[a.0.1421]dep 428[ExprUID] False[SSA]
==============End Node==============
[a.0.1422]push null
[a.0.1423]retb
```
 In case 2, they look like:
```
==============Start Node==============
[a.0.1414]push "one"
[a.0.1415]pop _lx
[a.0.1416]push _lx
[a.0.1417]pop var_0f7fcd7ccf20424f9230dc55d301231a
[a.0.1418]dep 428[ExprUID] False[SSA]
==============End Node==============
[a.0.1419]push null
[a.0.1420]retb
```

After the optimization to skip recompilation in #12835, execution should pick up from the instruction after the `pop _lx` in the update cycle for it to work, but owing to the AST being an identifier in case 1, and not a primitive, the program counter was not being set to that instruction (after the pop). 

The attempted fix is to avoid doing a static single assignment (SSA) transformation to convert something like `a = null;` to `temp_identifier = null; a = temp_identifier;`

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section**

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Engine/ProtoAssociative/CodeGen_SSA.cs [hunk 1]
```diff
@@ -488,7 +441,6 @@ private void DFSEmitSSA_AST(AssociativeNode node, Stack<AssociativeNode> ssaStac
 
                 var bnode = AstFactory.BuildAssignment(leftNode, rightNode);
                 bnode.isSSAAssignment = isSSAAssignment;
-                bnode.IsInputExpression = astBNode.IsInputExpression;
```
Comment (aparajit-pratap) [inline src/Engine/ProtoAssociative/CodeGen_SSA.cs:491]: This function is skipped from being called for ASTs that are "input" expressions.

File: src/Engine/ProtoAssociative/CodeGen_SSA.cs [hunk 2]
```diff
@@ -275,7 +275,7 @@ private List<AssociativeNode> BuildSSA(List<AssociativeNode> astList, ProtoCore.
                         BinaryExpressionNode bnode = (node as BinaryExpressionNode);
                         int generatedUID = ProtoCore.DSASM.Constants.kInvalidIndex;
 
-                        if (context.applySSATransform && core.Options.GenerateSSA)
+                        if (context.applySSATransform && core.Options.GenerateSSA && !bnode.IsInputExpression)
```
Comment (aparajit-pratap) [inline src/Engine/ProtoAssociative/CodeGen_SSA.cs:278]: Skip SSA for cases such as this:
```
a = null;
// update a
a = <some primitive>;
```

which would after SSA turn into this:
```
temp = null;
a = temp;
...
a = <some primitive>; // breaks!
```

Comment (mjkkirschner) [reply src/Engine/ProtoAssociative/CodeGen_SSA.cs:278]: do you think this is a good constraint to have on SSA, or should we mark this as a temporary workaround and file a task to fix the optimization in the future?

File: src/Engine/ProtoAssociative/CodeGen_SSA.cs (hunk 1 above)
Comment (mjkkirschner) [reply src/Engine/ProtoAssociative/CodeGen_SSA.cs:491]: If we ever intend to make SSA form work with this input optimization maybe we should leave this in, commented out with a TODO?

**Review by mjkkirschner — APPROVED** [review]

**Pull request merged**